src/NClient/NClient.Standalone/ClientProxy/Generation/Interceptors/ClientInterceptor.cs

```diff
@@ -148,32 +153,58 @@ internal class ClientInterceptor<TClient, TRequest, TResponse> : AsyncIntercepto
             }
         }
         
-        private async Task<object?> ExecuteHttpResponseAsync(ITransportNClient<TRequest, TResponse> transportNClient, IRequest request, Type resultType, IResiliencePolicy<TRequest, TResponse>? resiliencePolicy, CancellationToken cancellationToken)
+        private async Task<object?> ExecuteHttpResponseAsync(ITransportNClient<TRequest, TResponse> transportNClient, IRequest request, Type resultType, IResiliencePolicy<TRequest, TResponse>? resiliencePolicy, ICachingAttribute? cachingAttribute = default, CancellationToken cancellationToken = default)
         {
+            if (await _responseCacheWorker.TryGet(request, cancellationToken) is { } cachedResult)
```

Comment:
Will the cache work correctly if we have two methods that create the same query, but muppet the results differently? For example:
```C#
[GetMethod("entity")]
public Task<Entity> Get();
[GetMethod("entity")]
public Task<IResult<Entity>> Get();
```